### ISSUE TITLE
Trying to close already closed BindContext

### ISSUE THREAD

**Issue by patriczek** [fantix/gino#349]
* GINO version: 0.7.5
* Python version: 3.6.6
* asyncpg version: 0.17.0 
* aiocontextvars version: 0.1.2
* PostgreSQL version: postgres:10.3-alpine

### Description

When i run tests, then gino trying to close already closed bind
```

test/integration/test_add_extra_number_segment.py:26 (test_unable_find_route)
def finalizer():
        """Yield again, to finalize."""
        async def async_finalizer():
            try:
                await gen_obj.__anext__()
            except StopAsyncIteration:
                pass
            else:
                msg = "Async generator fixture didn't stop."
                msg += "Yield only once."
                raise ValueError(msg)
    
>       loop.run_until_complete(async_finalizer())

../../../venv/wisdom6/lib/python3.6/site-packages/pytest_asyncio/plugin.py:94: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/base_events.py:468: in run_until_complete
    return future.result()
../../../venv/wisdom6/lib/python3.6/site-packages/pytest_asyncio/plugin.py:86: in async_finalizer
    await gen_obj.__anext__()
conftest.py:23: in app
    yield client
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <gino.api._BindContext object at 0x10ae6b6d8>, exc_type = None
exc_val = None, exc_tb = None

    async def __aexit__(self, exc_type, exc_val, exc_tb):
>       await self._args[0].pop_bind().close()
E       AttributeError: 'NoneType' object has no attribute 'close'

../../../venv/wisdom6/lib/python3.6/site-packages/gino/api.py:183: AttributeError
```
But i'm not able to make script which will reproduce this error.
Below is code which will fix this error. If you would like i can make a PR with this change

### What I Did

Change from:
```
class _BindContext:
    async def __aexit__(self, exc_type, exc_val, exc_tb):
        await self._args[0].pop_bind().close()
```

to:
```
class _BindContext:
    async def __aexit__(self, exc_type, exc_val, exc_tb):
        bind_to_close = self._args[0].pop_bind()
        if bind_to_close:
             await bind_to_close.close()

```

**Comment by wwwjfy** [comment]
Thanks. One more check surely does no harm.
But I wonder how you met this issue? usually, you don't need to call `pop_bind` manually.

**Comment by patriczek** [comment]
In fact i have no idea. I got project after ex-stuff and this starts happens when i started write tests.
I'm currently trying find how this case happens and write script for reproduce.

**Comment by patriczek** [comment]
Finally i found it. 
In master it's solved.

In version 0.7.5  when you create a gino object is bind set on None and this is reason why it's failed
```
from gino.ext.aiohttp import Gino
g = Gino()
```

In master is bind set by default on "PlaceHolder" and this solved this issue as well.

**Comment by wwwjfy** [comment]
You mean it's closed before bind is set. I see.
Glad you found it.

**Comment by wwwjfy** [comment]
duplicate of #307, fixed in #310
